### PR TITLE
use wasm object format when '-shared' and 'side_module' are specified

### PR DESCRIPTION
**Pull request by nokotan**
This modification will enhance compatibility with cmake or auto-tool based projects, which add '-shared' flag when building shared libraries.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -1409,7 +1409,7 @@ def phase_linker_setup(options, state, newargs, settings_map):
     # of LTO.
     if final_suffix in EXECUTABLE_ENDINGS:
       diagnostics.warning('emcc', '-shared/-r used with executable output suffix. This behaviour is deprecated.  Please remove -shared/-r to build an executable or avoid the executable suffix (%s) when building object files.' % final_suffix)
-    else:
```
Comment (sbc100) [inline emcc.py:1412]: This change LGTM.. can we land this without the `cmake/Modules/Platform/Emscripten.cmake ` change though.  That looks separate and perhaps less obviously better.

Can you add a test for this too?  Maybe looks for the existing test for warning in question (`'linking a library with `-shared` will emit a static object file`) and modify that test?

Comment (nokotan) [reply emcc.py]: Ok, I'll move changes in `cmake/Modules/Platform/Emscripten.cmake` into another PR.

**Review by sbc100 — APPROVED** [review]
Nice!  Thanks for working on this.

**Comment by sbc100** [comment]
The test failures here look unrelated.  I believe they should go away if you rebase/merge with `origin/main`.

**Comment by kripken** [comment]
This looks ready to land I think, unless you had more issues @sbc100 ?

**Comment by kripken** [comment]
(this had automerge, but looks like that was removed)

File: tests/test_other.py [hunk 2]
```diff
@@ -10136,6 +10139,18 @@ def test_shared_flag(self):
     self.assertContained('warning: -shared/-r used with executable output suffix', err)
     self.run_js('out.js')
 
+  def test_shared_and_shared_module_flag(self):
```
Comment (sbc100) [inline tests/test_other.py:10142]: I think this should be `test_shared_and_side_module_flag` or maybe just `test_shared_and_shared_module`?

**Review by sbc100 — APPROVED** [review]
lgtm with test name fix

File: tests/test_other.py (hunk 2 above)
Comment (nokotan) [reply tests/test_other.py]: oh, thanks for pointing this out

**Pull request merged**